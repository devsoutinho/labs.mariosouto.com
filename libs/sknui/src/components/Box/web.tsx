import React from "react";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface BoxProps {
  tag?: string;
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  // receive anything
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}
export function Box({ tag, styleSheet, children, ...props }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Tag = tag || ("div" as any);
  return (
    <Tag
      style={{
        display: "flex",
        flexDirection: "column",
        ...styleSheet,
      }}
      {...props}
    >
      {children}
    </Tag>
  );
}
