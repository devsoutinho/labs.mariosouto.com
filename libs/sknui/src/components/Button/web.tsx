import React from "react";
import { useTheme } from "../../theme/web";
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Text, Touchable } from "../web";

interface ButtonProps {
  type?: "button" | "submit" | "reset";
  variant: "primary" | "secondary" | "accent" | "ghost" | "destructive";
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
}
export function Button({
  variant,
  styleSheet,
  children,
  prefix,
  suffix,
}: ButtonProps) {
  const { theme } = useTheme();
  const PrefixComponent = prefix;
  const SuffixComponent = suffix;

  const stylesByVariant = {
    primary: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.primary.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.primary.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    secondary: {
      color: theme.colors.neutral.x500,
      backgroundColor: theme.colors.neutral.x000,
      borderStyle: "solid",
      borderWidth: theme.spacing.x_unit,
      borderColor: theme.colors.neutral.x300,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.neutral.x300} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    accent: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.accent.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.accent.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    ghost: {
      color: theme.colors.primary.x500,
      backgroundColor: theme.colors.neutral.x000,
      boxShadow: theme.shadow.none,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.primary.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
    destructive: {
      color: theme.colors.neutral.x000,
      backgroundColor: theme.colors.negative.x500,
      boxShadow: theme.shadow.default,
      _focusVisible: {
        boxShadow: `${theme.colors.neutral.x000} 0px 0px 0px 2px, ${theme.colors.negative.x500} 0px 0px 0px 4px, rgba(0, 0, 0, 0.05) 0px 1px 2px 0px`,
      },
    },
  };

  const currentVariant = stylesByVariant[variant];
  const { backgroundColor } = stylesByVariant[variant];

  const buttonStyles = {
    borderWidth: theme.spacing.x_unit,
    borderColor: "transparent",
    ...currentVariant,
    backgroundColor,
    overflow: "initial",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    flex: 1,
    appearance: "none",
    gap: "8px",
    borderRadius: theme.rounded.md,
    paddingY: theme.spacing.x2,
    paddingX: theme.spacing.x4,
    _placeholder: {
      color: theme.colors.neutral.x400,
    },
    _hover: {
      opacity: 0.8,
    },
    _focusVisible: {
      ...currentVariant?._focusVisible,
      outline: `2px solid transparent`,
      outlineOffset: "2px",
    },
  };

  return (
    <Touchable
      styleSheet={{
        ...styleSheet,
        ...(buttonStyles as unknown as StyleSheet),
      }}
    >
      {PrefixComponent}
      <Text>{children}</Text>
      {SuffixComponent}
    </Touchable>
  );
}

Button.defaultProps = {
  variant: "primary",
};
