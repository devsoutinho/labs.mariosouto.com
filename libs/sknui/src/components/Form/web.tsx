import React from 'react';
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Box } from "../web";

interface FormProps {
  styleSheet?: StyleSheet;
  children: React.ReactNode;
}
export function Form({ styleSheet, children }: FormProps) {
  const tag = "form";

  return <Box tag={tag} styleSheet={{
    ...styleSheet,
  }}>
    {children}
  </Box>;
}
