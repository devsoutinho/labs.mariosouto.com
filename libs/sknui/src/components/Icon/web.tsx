import React from 'react';
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Box } from "../web";

interface IconProps {
  name?: string;
  styleSheet?: StyleSheet;
}
export function Icon({ name, styleSheet }: IconProps) {
  return (
    <Box styleSheet={styleSheet}>
      {name}
    </Box>
  );
}
