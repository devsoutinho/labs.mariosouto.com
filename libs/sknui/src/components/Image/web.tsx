import { Box } from "../Box/web";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface ImageProps {
  src?: string;
  alt: string;
  styleSheet?: StyleSheet;
}
export function Image({ src, alt, styleSheet }: ImageProps) {
  const { objectFit, objectPosition, borderRadius, ...pictureStyleSheet } =
    styleSheet || {};

  return (
    <Box
      tag="picture"
      styleSheet={{
        ...pictureStyleSheet,
        overflow: "hidden",
        borderRadius,
      }}
    >
      <Box
        tag="img"
        src={src}
        alt={alt}
        style={{
          overflow: "hidden",
          borderRadius,
          objectPosition,
          objectFit: objectFit || "cover",
          width: "100%",
          height: "100%",
        }}
      />
    </Box>
  );
}
