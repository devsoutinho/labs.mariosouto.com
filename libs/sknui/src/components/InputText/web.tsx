import React from "react";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface InputTextProps {
  label?: string;
  styleSheet?: StyleSheet;
}
export function InputText({ label, styleSheet }: InputTextProps) {
  return (
    <label>
      {label}
      <input
        type="text"
        style={
          {
            display: "flex",
            ...styleSheet,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
          } as any
        }
      />
    </label>
  );
}
