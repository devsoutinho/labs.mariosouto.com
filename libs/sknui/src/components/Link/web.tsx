import React from 'react';
import { StyleSheet } from "../../theme/StyleSheet/web";
import { Touchable } from "../web";

interface LinkProps {
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  prefix?: React.ReactNode;
  suffix?: React.ReactNode;
  href: string;
}
export function Link({ styleSheet, children, href, prefix, suffix, ...props }: LinkProps) {
  const PrefixComponent = prefix;
  const SuffixComponent = suffix;

  return <Touchable
    href={href}
    styleSheet={{
      display: "flex",
      ...styleSheet,
    }}
    {...props}>
    {PrefixComponent}
    {children}
    {SuffixComponent}
  </Touchable>;
}
