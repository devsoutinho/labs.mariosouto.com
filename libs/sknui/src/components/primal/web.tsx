import React from "react";
import { Box as ChakraBox } from "@chakra-ui/react";
import { StyleSheet } from "../../theme/StyleSheet/web";

interface BoxProps {
  tag?: string;
  styleSheet?: StyleSheet;
  children: React.ReactNode;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}
export function BoxPrimal({ tag, styleSheet, children, ...props }: BoxProps) {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Tag = tag || ("div" as any);
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const Comp = ChakraBox as any;

  const overflow = "hidden";

  return (
    <Comp as={Tag} overflow={overflow} {...styleSheet} {...props}>
      {children}
    </Comp>
  );
}
