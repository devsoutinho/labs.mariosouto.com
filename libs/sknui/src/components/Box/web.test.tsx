import React from "react";
import { Box } from "./web";
import { customRender } from "../../../commons/testing-library/web";
import { Text } from "../Text/web";

const render = customRender();

describe("<Box />", () => {
  it("renders as expected", () => {
    const { container } = render(
      <Box>
        <Text>Hello</Text>
      </Box>
    );

    expect(container).toMatchSnapshot();
  });
});
