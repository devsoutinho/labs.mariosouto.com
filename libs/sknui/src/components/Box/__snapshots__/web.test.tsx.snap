// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders as expected 1`] = `
<div>
  <div
    style="display: flex; flex-direction: column;"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;
